tests/framework/framework.go

```diff
@@ -76,34 +80,42 @@ func init() {
 // NewFramework makes a new framework and sets up the global BeforeEach/AfterEach's.
 // Test run-time flags are parsed and added to the Framework struct.
 func NewFramework(prefix string) *Framework {
+	var err error
+
 	f := &Framework{
 		NsPrefix: prefix,
 	}
 
 	// handle run-time flags
 	if !flag.Parsed() {
 		flag.Parse()
-		// report flags values passed to test binary
-		fmt.Fprintf(GinkgoWriter, "** Test flags:\n")
-		flag.Visit(func(f *flag.Flag) {
-			fmt.Fprintf(GinkgoWriter, "   %s = %q\n", f.Name, f.Value.String())
-		})
-		fmt.Fprintf(GinkgoWriter, "**\n")
-		if kubectlPath != nil {
-			f.KubectlPath = *kubectlPath
-		}
-		if ocPath != nil {
-			f.OcPath = *ocPath
-		}
-		if cdiInstallNs != nil {
-			f.CdiInstallNs = *cdiInstallNs
-		}
-		if kubeConfig != nil {
-			f.KubeConfig = *kubeConfig
-		}
-		if master != nil {
-			f.Master = *master
-		}
+	}
+	// report flags values passed to test binary
+	fmt.Fprintf(GinkgoWriter, "** Test flags:\n")
+	flag.Visit(func(f *flag.Flag) {
+		fmt.Fprintf(GinkgoWriter, "   %s = %q\n", f.Name, f.Value.String())
+	})
+	fmt.Fprintf(GinkgoWriter, "**\n")
+	if kubectlPath != nil {
+		f.KubectlPath = *kubectlPath
+	}
+	if ocPath != nil {
+		f.OcPath = *ocPath
+	}
+	if cdiInstallNs != nil {
+		f.CdiInstallNs = *cdiInstallNs
+	}
+	if kubeConfig != nil {
+		f.KubeConfig = *kubeConfig
+	}
+	if master != nil {
+		f.Master = *master
+	}
+
+	f.RestConfig, err = f.LoadConfig()
+	if err != nil {
+		// Can't use Expect here due to fail handler not being registered at this point
```

Comment:
when is the Fail handler registered? I thought it was in the zz_test_suite.go code, which is run prior to `NewFramework`

Reply:
@jeffvance I actually put the Expect in there, and it errored out with a message saying the Fail handler wasn't registered yet. I was shocked at this, so I put a note in there to remind me why I didn't put an Expect in there.

Reply:
panic: You are trying to make an assertion, but Gomega's fail handler is nil.
If you're using Ginkgo then you probably forgot to put your assertion in an It().
Alternatively, you may have forgotten to register a fail handler with RegisterFailHandler() or RegisterTestingT().

That is the error message. So I think the problem is that we are calling the Expect outside of the It block.

Reply:
Agree. That must be the issue here.